### PR TITLE
Implements Hint Features. Docker Dev Environment an Option Now. UI/UX issues fixed

### LINKED ISSUE #2: Add hints
It would be nice to show some hints for a person, so if you aren't sure you're not just faced with an empty text box.

I was imagining something like "It's one of these four names", and if you click hint again "It's one of these two names".

### LINKED ISSUE #6: Compare case insensitively
Find a library or something that allows case-insensitive and diacritic-insensitive comparison, so that typing 'foo' would match a person named 'Foo' and typing 'bar' would match a person named 'bâr'.

### LINKED ISSUE #10: Focus on text area on load and after next clicked
Currently nothing is focused.

### LINKED ISSUE #11: If Guess button is clicked, focus does not return to text area
If the Enter key is pressed, the focus stays on the text area which is the desired behavior.

### LINKED ISSUE #12: If guess is correct, focus should move to Next key automatically?
Not sure if this should be the preferred interaction but think so. Thoughts?

### LINKED ISSUE #17: Hint should be disabled after 1 click

### PR DESCRIPTION
**Pull request by botwhytho**
Can use 'docker-compose up' from root folder of repo to build and set-up all environments.

An actual .sql dump would need to be in the .init.d/db/SQL folder before running the docker command, but .gitignore is correctly excluding that.

Could not resolve changing proxy values in package.json by any combination of environmental variables or command line flags so leaving it as localhost for the commit.

Implements:
Implements #2 

Resolves:
Resolves #6 
Resolves #10 
Resolves #11 
Resolves #12 
Resolves #17

### CODE REVIEW

File: src/components/hint.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+class Hint extends Component {
+  render() {
+    const hint = this.props.hint;
+    let persons = this.props.person.slice();
+    if (hint.get('status') === "yes please") {
+      let people = "REACT_APP_USE_TEST_DATA" in process.env
+      ? persons
+          .sort(function(a,b) {
+            return Math.random()-Math.random();
```
Comment (jasonaowen) [inline src/components/hint.js]: In our discussion, we were having a hard time figuring out if this does what we want. Is there a more conceptually simple way to express this?

File: src/components/card.js [hunk 2]
```diff
@@ -3,37 +3,57 @@ import { connect } from 'react-redux';
...
+      this.props.hint();
+    }
   }
 
   render() {
     const person = this.props.person;
     return (
       <div className="card">
-        <div className="person-title" id={person.get('person_id') + "-title"}>
+        <div className="person-title" id={person.getIn(['person_id']) + "-title"}>
```
Comment (jasonaowen) [inline src/components/card.js]: Should this have a `0` in the `getIn` path?

File: src/components/hint.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+
+class Hint extends Component {
+  render() {
+    const hint = this.props.hint;
+    if (hint.get('status') === "yes please") {
+      let people = this.props.person.slice().sort(function(a,b) {
+        return Math.random() - Math.random();
```
Comment (jasonaowen) [inline src/components/hint.js]: In our discussion, we were having a hard time figuring out if this does what we want. Is there a more conceptually simple way to express this?

**Comment by jasonaowen** [comment]
:+1: This looks good to me!

**Comment by botwhytho** [comment]
Fixed not-really-random sort of hint names. Fixed div id issue.

**Pull request merged**